Update to 22.08 runtime, and use our own libs

tinywrkb opened the pull request:
* Bump to the latest 21.08 Freedesktop runtime
* Use runtime's libs as much as possible, and build from source what's still needed.
* Set interpreter with patchelf to the runtime's one
* Drop unneeded wrapper script
* Unset `QT_QPA_PLATFORM` to avoid having the app break when user set it
  globally to wayland. This is not an issue anymore with KDE Plasma, but it's a
  good service to the user
* Add f-e-d-c properties, and drop the unsupported JSON comment which
  will be removed by f-e-d-c

This needs to be tested to confirm that nothing is broken. As far as I can tell, it's working correctly.

tinywrkb commented:
I'm not sure if libbsd is actually needed. I added it as the app ship it, but at least when opening an image and extracting files, libbsd is not being used.

tinywrkb commented:
/ping @hadess

hadess commented:
Thanks for the patches and updating the PR. Can you please split up the changes further, in a way that the package stays buildable, but the patches are small and trivial?

tinywrkb committed:
Add automated version check

Using flatpak-external-data-checker.

tinywrkb committed:
Set interpreter with patchelf to the runtime's one

Otherwise poweriso-gui won't start:
$ poweriso-gui
bash: /app/bin/poweriso-gui: No such file or directory

tinywrkb committed:
Use FLATPAK_ID and FLATPAK_DEST

tinywrkb committed:
Add missing dependencies for PowerISO's Qt5 build

As we use the Qt5 build that comes with this piece of software, we need
to build older versions of some libraries to satisfy its dependencies.

tinywrkb committed:
Require important update to open PRs

tinywrkb committed:
Drop unneeded wrapper script

tinywrkb committed:
Unset QT_QPA_PLATFORM

Avoid having the app break when user globally sets it to "wayland".

This is not an issue anymore with KDE Plasma, which previously exported
the variable, but it's a good service to the user who might not be aware
that it's wrong and unneeded to set it globally.

tinywrkb commented:
@hadess sorry for the late reply. I think this should do it.

hadess commented on com.poweriso.PowerISO.json:
```diff
@@ -8,8 +8,9 @@
-        "--filesystem=home"
+        "--env=QT_QPA_PLATFORM_PLUGIN_PATH=/app/lib",
+        "--filesystem=home",
+        "--socket=x11"
```
Please don't reorder items that don't need to be.

hadess commented on com.poweriso.PowerISO.json:
```diff
@@ -7,10 +7,11 @@
         "--env=QT_QPA_PLATFORM_PLUGIN_PATH=/app/lib",
         "--filesystem=home",
-        "--socket=x11"
+        /* Only supports X11 with the builtin Qt */
```
Again with the reordering.

hadess commented on com.poweriso.PowerISO.json:
```diff
@@ -9,7 +9,6 @@
         "--device=all",
         "--env=QT_QPA_PLATFORM_PLUGIN_PATH=/app/lib",
         "--filesystem=home",
-        /* Only supports X11 with the builtin Qt */
```
It's unsupported by flatpak-external-data-checker. It's supported by Flatpak, and will work just fine. Please leave it as-is.

hadess commented on com.poweriso.PowerISO.json:
```diff
@@ -12,7 +12,98 @@
+    ],
     "modules": [
+        {
+            "name": "icu",
```
Please explain in the commit message why we need to compile those modules ourselves. I would also have preferred separate commits for each one of the modules.

hadess committed:
Update runtime to 22.08

Closes: #3

hadess committed:
Don't require /app/lib not to exist

It will be created by the patchelf installation, but as it's better not
to rely on another module creating our directories, pass "-p" to mkdir.
